Objects/dictobject.c

```diff
@@ -459,23 +459,26 @@ static PyObject *empty_values[1] = { NULL };
 int
 _PyDict_CheckConsistency(PyObject *op, int check_content)
 {
-#ifndef NDEBUG
-    _PyObject_ASSERT(op, PyDict_Check(op));
+#define ASSERT(expr) \
```

Comment:
I'm not particularly fond of 'assert' and 'ASSERT' doing different things, especially since ASSERT does not use assert and does not do nothing in NDEBUG mode. How about 'CHECK' instead?